azurerm/resource_arm_storage_account.go

```diff
@@ -909,6 +918,11 @@ func expandStorageAccountNetworkRules(d *schema.ResourceData) *storage.NetworkRu
 	networkRule := networkRules[0].(map[string]interface{})
 	networkRuleSet := &storage.NetworkRuleSet{}
 
+	if networkRule["default_action"].(string) == "Allow" {
```

Comment:
we should probably make this:

```suggestion
	if networkRule["default_action"].(string) == string(storage.DefaultActionAllow) {
```